eos/form-factors/analytic-d-to-pi.cc

```diff
@@ -0,0 +1,2177 @@
+            const double M2_rescaled = this->M2() * this->rescale_factor_p(q2);
+            const double fB = decay_constant();
+            const double F_lo = F_lo_tw2(q2, M2_rescaled) + F_lo_tw3(q2, M2_rescaled) + F_lo_tw4(q2, M2_rescaled);
+            const double F_nlo = F_nlo_tw2(q2, M2_rescaled) + F_nlo_tw3(q2, M2_rescaled);
+            /*
+             * we estimate the NNLO corrections to obey the relation |F_nnlo / F_nlo| = |F_nlo / F_lo|.
+             * Therefore we set F_nnlo = F_nlo^2 / F_lo * zeta_nnlo, where zeta ranges between -1 and +1.
+             */
+            const double F_nnlo = F_nlo * F_nlo / F_lo * zeta_nnlo;
+            const double alpha_s = model->alpha_s(mu);
+
+            return std::exp(MD2 / M2_rescaled) / (2.0 * MD2 * fB) * (F_lo + alpha_s / (3.0 * M_PI) * F_nlo + alpha_s * alpha_s / (9.0 * M_PI * M_PI) * F_nnlo);
+        }
+
+        double f_0(const double & q2) const
+        {
+
+            if (std::abs(q2) < 1e-6)
+                return f_p(q2);
+
+            const double MD2 = MD * MD, mpi2 = mpi * mpi;
+            const double M2_rescaled = this->M2() * this->rescale_factor_0(q2);
+            const double fB = decay_constant();
+            const double F_lo = F_lo_tw2(q2, M2_rescaled) + F_lo_tw3(q2, M2_rescaled) + F_lo_tw4(q2, M2_rescaled);
+            const double F_nlo = F_nlo_tw2(q2, M2_rescaled) + F_nlo_tw3(q2, M2_rescaled);
+            const double Ftil_lo = Ftil_lo_tw3(q2, M2_rescaled) + Ftil_lo_tw4(q2, M2_rescaled);
+            const double Ftil_nlo = Ftil_nlo_tw2(q2, M2_rescaled) + Ftil_nlo_tw3(q2, M2_rescaled);
+            //const double Ftil_nnlo = F_nlo * F_nlo / F_lo * zeta_nnlo;
+            const double alpha_s = model->alpha_s(mu);
+
+            return std::exp(MD2 / M2_rescaled) / (2.0 * MD2 * fB) * (2.0 * q2 / (MD2 - mpi2) * (Ftil_lo + alpha_s / (3.0 * M_PI) * Ftil_nlo) + (1.0 - q2 / (MD2 - mpi)) * (F_lo + alpha_s / (3.0 * M_PI) * F_nlo));
+        }
+
+        double f_t(const double & q2) const
+        {
+            const double MD2 = MD * MD;
+            const double M2_rescaled = this->M2() * this->rescale_factor_T(q2);
+            const double fB = decay_constant();
+            const double FT_lo = FT_lo_tw2(q2, M2_rescaled) + FT_lo_tw3(q2, M2_rescaled) + FT_lo_tw4(q2, M2_rescaled);
+            const double FT_nlo = FT_nlo_tw2(q2, M2_rescaled) + FT_nlo_tw3(q2, M2_rescaled);
+            //const double FT_nnlo = FT_nlo * FT_nlo / FT_lo * zeta_nnlo;
+            const double alpha_s = model->alpha_s(mu);
+
+            return std::exp(MD2 / M2_rescaled) / (2.0 * MD2 * fB) * (MD + mpi) * (FT_lo + alpha_s / (3.0 * M_PI) * FT_nlo);
+        }
+
+        Diagnostics diagnostics() const
+        {
+            Diagnostics results;
+
+            // Function rho_1, cf. [DKMMO2008], eq. (C.2)
+            {
+                results.add(Diagnostics::Entry{ rho_1(19.60, 4.16, 4.16), "rho_1(s = 19.60, m_b = 4.16, mu = 4.16), [DKMMO2008]" });
```

Comment:
Please
 - add reference for https://arxiv.org/pdf/0907.2842.pdf as KKMO:2009A (if it does not already exist), in a new commit.
 - change to ``s = {6.5, 7, 7.5}, m_c = 1.27, mu = 1.4``, and refer to [KKMO:2009A] rather than DKMMO2008.